lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -168,32 +178,65 @@ def _get_account_info(self, username_or_email):
         and program_enrollments_info. If we cannot identify the user, return
         empty object and error.
         """
-        user_info = {}
-        external_key = None
         try:
             user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
-            user_info['username'] = user.username
-            user_info['email'] = user.email
+            user_social_auth = None
             try:
                 user_social_auth = UserSocialAuth.objects.get(user=user)
-                _, external_key = user_social_auth.uid.split(':', 1)
-                user_info['external_user_key'] = external_key
-                user_info['SSO'] = {
-                    'uid': user_social_auth.uid,
-                    'provider': user_social_auth.provider
-                }
             except UserSocialAuth.DoesNotExist:
                 pass
-
+            user_info = self._serialize_user_info(user, user_social_auth)
             enrollments = self._get_enrollments(user=user)
             result = {'user': user_info}
             if enrollments:
                 result['enrollments'] = enrollments
 
+            result['id_verification'] = IDVerificationService.user_status(user)
             return result, ''
         except User.DoesNotExist:
             return {}, 'Could not find edx account with {}'.format(username_or_email)
 
+    def _get_external_user_info(self, external_user_key, idp_slug):
+        """
+        Provided the external_user_key and idp_slug, return edx account info
+        and program_enrollments_info if any. If we cannot identify the data,
+        return empty object.
+        """
+        sso_uid = ':'.join([idp_slug, external_user_key])
+        user_social_auths = UserSocialAuth.objects.filter(
+            uid=sso_uid
+        ).select_related('user').order_by('-id')
+        result = {}
+
+        if user_social_auths:
+            user_social_auth = user_social_auths.first()
+            user = user_social_auth.user
+            user_info = self._serialize_user_info(user, user_social_auth)
+            result['user'] = user_info
+            result['id_verification'] = IDVerificationService.user_status(user)
+        enrollments = self._get_enrollments(external_user_key=external_user_key)
+        if enrollments:
+            result['enrollments'] = enrollments
+        return result
```

Comment:
I'm sure this works correctly, but I have concern about any support code that re-implements business logic instead of just calling out to the same functions that the real application APIs use.

For example, imagine we tweak the logic in `lms/djangoapps/program_enrollments` that calculates how users are associated with external keys. If we forget that this logic also needs to be changed in `lms/djangoapps/support`, then all of a sudden the results that the Support Tool returns will be out of sync with the results that the Program Enrollments API returns, which is dicey.

In this specific case, program_enrollments has [`get_users_by_external_keys`](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/program_enrollments/api/reading.py#L345), which doesn't exactly have the call signature you need, but you could refactor out a function that takes an `org_key` and a set of `external_user_keys` in order to keep the business logic all in one place.

Reply:
I took in this feedback and refactored the `reading.py` functions to fit the needs of the support tool.